### ISSUE TITLE
Add a vm.iter cheatecode

### ISSUE THREAD

**Issue by 1anyway** [foundry-rs/foundry#4858]
### Component

Forge

### Describe the feature you would like

In my development practice, i came across scenarios where I had to iterate through all possible values, but i found that Foundry didn't provide the tools or cheatcodes. So I had to do it that way: 
```
address[] private tokens = [WBTC, USDC, USDT, UNI, LINK];
for (uint256 _index = 0; _index < tokens.length; ++_index) {
            deal(tokens[_index], alice, assetAmount);
            
            IERC20(tokens[_index]).approve(address(gloopZapper), assetAmount);
            gloopZapper.mintWGLP(
                tokens[_index], 
                assetAmount,
                0,
                0
            );
```
It would be greater to provide a cheatecode like `vm.iter()` to iterate through all values.

**Comment by mds1** [comment]
Can you expand on how this cheatcode would work with some examples? I'm not sure I understand what this would do that you can't do with a regular solidity `for` loop?

**Comment by 1anyway** [comment]
Yes, it can be finished by `for` loop. But if there is a iterator to help us to do it, foundry will become more convenient and powerful

**Comment by mds1** [comment]
I still don't understand how this would look or work, given the limitations of solidity. I'm going to close this as a result, but can reopen if you share a more detailed proposal :)

**Comment by 1anyway** [comment]
This is the using example 
```
address[] private tokens = [WBTC, USDC, USDT, UNI, LINK];

function test(address _token) public {
  _token = iter(_token, tokens);
  deal(_token, alice, assetAmount);
  IERC20(_token).approve(user, assetAmount);
  // Do sth else
}
```
It's kind of like `vm.assume` and `bound` in foundry, but the difference is `vm.assume` and `bound` only run a limited number of scenarios in fuzzy tests, I hope to have a cheatcode `vm.iter()` to iterate through all discrete values in an array.

**Comment by 1anyway** [comment]
Yes, it runs each discrete value in the array once. It's similar to #858 and #4839, but I guess not exactly.

**Comment by 1anyway** [comment]
If all type combinations become overloaded, you can divide them into different types, For example, `vm.iter_Address()`, `vm.iter_Uint()` and so on. I think this is more user friendly than requesting an ABI.

**Comment by zerosnacks** [comment]
Thanks for your suggestion, marking as `not planned` per discussion above as it is currently not within scope and effectively superseded by `fuzz fixtures`: https://book.getfoundry.sh/forge/fuzz-testing
